components/FirebaseAuth.js

```diff
@@ -0,0 +1,51 @@
+/* globals window */
```

Comment:
this handles firebase authentication, boiler plate code from NextJS's own tutorial